src/model/entities/rundown.ts

```diff
@@ -564,17 +564,21 @@ export class Rundown extends BasicRundown {
       return
     }
 
+    // We know that 'segmentToRemove' is not undefined at this point
+    this.segments = this.segments.filter(segment => segment.id !== segmentId)
```

Comment:
You could remove the comment by using the `segmentToRemove.id` to check against. It should have the same outcome.

Reply:
It was to avoid a possibly undefined id.